src/utils/discord.ts

```diff
@@ -0,0 +1,16 @@
+import { Message, MessageCreateOptions } from 'discord.js'
+
+/**
+ * Sends a direct message (DM) to the author of a message if they have enabled DMs.
+ *
+ * @param {Message} message - The original message object.
+ * @param {MessageCreateOptions} payload - The payload of the message to be sent.
+ */
+export async function sendDm(
+  message: Message,
+  payload: MessageCreateOptions,
+): Promise<void> {
+  if (message.author.dmChannel) {
+    message.author.send(payload)
```

Comment:
fix indent krub
```suggestion
  try {
    const dm = await message.author.createDM()
    await dm.send(payload)
  } catch (error) {
    console.error(`Could not send DM to user ${message.author.id}: ${error}`)
```

Reply:
look like catch error is the only way we can check if user can receive dm.
We might check for error code 50007 like in [original suggestion](https://github.com/kaogeek/kaogeek-discord-bot/pull/140#issuecomment-1562882379), and log for error only when message !== 50007 (Cannot send messages to this user)

ref on error code: https://discord.com/developers/docs/topics/opcodes-and-status-codes